pkg/apis/rollouts/v1alpha1/analysis_types.go

```diff
@@ -220,3 +221,41 @@ type Measurement struct {
 	// ResumeAt is the  timestamp when the analysisRun should try to resume the measurement
 	ResumeAt *metav1.Time `json:"resumeAt,omitempty"`
 }
+
+type KayentaMetric struct {
+
+	Address string `json:"address"`
+
+	Application string `json:"application"`
+
+	CanaryConfigName string `json:"canaryConfigName,omitempty"`
+	CanaryConfigId   string `json:"canaryConfigId,omitempty"`
+
+	MetricsAccountName string `json:"metricsAccountName"`
+	ConfigurationAccountName string `json:"configurationAccountName"`
+	StorageAccountName  string `json:"storageAccountName"`
+
+
+	Threshold KayentaThreshold `json:"threshold"`
+
+	Scopes  []KayentaScope `json:"scopes"`
+}
+
+type KayentaThreshold struct {
+	Pass int `json:"pass,omitempty"`
+	Marginal int `json:"marginal,omitempty"`
+}
+
+type KayentaScope struct {
+	Name string `json:"name,omitempty"`
+	ControlScope ScopeDetail `json:"controlScope,omitempty"`
+	ExperimentScope ScopeDetail  `json:"experimentScope,omitempty"`
+}
+
+type ScopeDetail struct {
+	Scope string `json:"scope,omitempty"`
+	Region string `json:"region,omitempty"`
+	Step int `json:"step,omitempty"`
+	StartTime string `json:"start,omitempty"`
+	EndTime string `json:"end,omitempty"`
+}
```

Comment:
I think we should make sure that we are marking the correct fields optional. For example, I don't think `Pass` should be optional. We can talk about this irl.

Reply:
fixed.